utility/src/main/java/org/oppia/util/parser/UrlImageParser.kt

```diff
@@ -78,9 +82,25 @@ class UrlImageParser private constructor(
     }
   }
 
-  private fun calculateInitialMargin(drawableWidth: Int): Int {
-    val availableAreaWidth = htmlContentTextView.width
-    return (availableAreaWidth - drawableWidth) / 2
+  private fun calculateInitialMargin(availableAreaWidth: Int, drawableWidth: Int): Int {
+    val margin = (availableAreaWidth - drawableWidth) / 2
+    return if (margin > 0) {
+      margin
+    } else {
+      0
+    }
+  }
+
+  // Reference: https://stackoverflow.com/a/51865494
+  fun <T : View> T.width(calculateWidth: (Int) -> Unit) {
```

Comment:
Not keen on this extension function because it has a high hidden cost, and potentially leaks a OnGlobalLayoutListener.

Reply:
Also, please name this something closer to what it's doing, e.g. 'computeWidthOnGlobalLayout'

Reply:
Update the code like this:
```
private fun TextView.width(computeWidthOnGlobalLayout: (Int) -> Unit) {
    if (width == 0) {
      viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
        override fun onGlobalLayout() {
          viewTreeObserver.removeOnGlobalLayoutListener(this)
          computeWidthOnGlobalLayout(width)
        }
      })
    } else {
      computeWidthOnGlobalLayout(width)
    }
  }
```

@BenHenning does this look correct? If it is correct, then you can merge this directly.